cpp/src/parquet/bloom_filter_benchmark.cc

```diff
@@ -0,0 +1,230 @@
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "benchmark/benchmark.h"
+
+#include "parquet/bloom_filter.h"
+
+#include <random>
+
+namespace parquet {
+namespace benchmark {
+
+constexpr static uint32_t kBloomFilterElementSize = 1024;
+
+std::unique_ptr<BloomFilter> createBloomFilter(uint32_t elementSize) {
+  std::unique_ptr<BlockSplitBloomFilter> block_split_bloom_filter =
+      std::make_unique<BlockSplitBloomFilter>();
+  block_split_bloom_filter->Init(
+      BlockSplitBloomFilter::OptimalNumOfBytes(elementSize, /*fpp=*/0.05));
+  std::unique_ptr<BloomFilter> bloom_filter = std::move(block_split_bloom_filter);
+  ::benchmark::DoNotOptimize(bloom_filter);
+  return bloom_filter;
+}
+
+constexpr static uint32_t kGenerateBenchmarkDataStringLength = 8;
+
+void GenerateRandomString(uint32_t length, std::vector<uint8_t>* heap) {
+  // Character set used to generate random string
+  const std::string charset =
+      "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz";
+
+  std::default_random_engine gen(0);
+  std::uniform_int_distribution<uint32_t> dist(0, static_cast<int>(charset.size() - 1));
+
+  for (uint32_t i = 0; i < length; i++) {
+    heap->push_back(charset[dist(gen)]);
+  }
+}
+
+template <typename T>
+void GenerateBenchmarkData(uint32_t size, T* data,
+                           [[maybe_unused]] std::vector<uint8_t>* heap) {
+  if constexpr (std::is_integral_v<T>) {
+    std::default_random_engine gen(/*seed*/ 0);
+    std::uniform_int_distribution<T> d(std::numeric_limits<T>::min(),
+                                       std::numeric_limits<T>::max());
+    for (uint32_t i = 0; i < size; ++i) {
+      data[i] = d(gen);
+    }
+  } else if constexpr (std::is_floating_point_v<T>) {
+    std::default_random_engine gen(/*seed*/ 0);
+    std::uniform_real_distribution<T> d(std::numeric_limits<T>::lowest(),
+                                        std::numeric_limits<T>::max());
+    for (uint32_t i = 0; i < size; ++i) {
+      data[i] = d(gen);
+    }
+  } else if constexpr (std::is_same_v<FLBA, T>) {
+    GenerateRandomString(kGenerateBenchmarkDataStringLength * size, heap);
+    for (uint32_t i = 0; i < size; ++i) {
+      data[i].ptr = heap->data() + i * kGenerateBenchmarkDataStringLength;
+    }
+  } else if constexpr (std::is_same_v<ByteArray, T>) {
+    GenerateRandomString(kGenerateBenchmarkDataStringLength * size, heap);
+    for (uint32_t i = 0; i < size; ++i) {
+      data[i].ptr = heap->data() + i * kGenerateBenchmarkDataStringLength;
+      data[i].len = kGenerateBenchmarkDataStringLength;
```

Comment:
Can we vary the string length somehow?

Reply:
It's ok, but I guess keep size same is no problem

Reply:
Probably not, though it may makes things easier for the CPU.

Reply:
Emmm should I use interleave values like `kGenerateBenchmarkDataStringLength - 1, kGenerateBenchmarkDataStringLength + 1` ... Or using other ways? I guess making FLBA and ByteArray have different sum bytes size is unfair

Reply:
Good point. We may keep things like that as it's probably good enough.